internal/services/mysql/mysql_flexible_server_resource.go

```diff
@@ -267,6 +267,11 @@ func resourceMysqlFlexibleServer() *pluginsdk.Resource {
 							Computed:     true,
 							ValidateFunc: validation.IntBetween(20, 16384),
 						},
+						"io_scaling_enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  true,
```

Comment:
Is this default documented/Is that how the API is behaving?

Reply:
As far as i can tell, the default value is `Disabled` in the [API specs](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/stable/2022-01-01/FlexibleServers.json).

The same is true for "auto_grow_enabled", where the default value in the provider is `True`/`Enabled`, but default value of API is `Disabled`

Reply:
Seeing as this property conflicts with `iops` it's problematic to default this to `true`. If the default in the API is `false` we should set that as the default here as well.

Reply:
Cool. Changed the default to `false`